CMakeLists.txt

```
@@ -0,0 +1,136 @@
+cmake_minimum_required(VERSION 3.0)
+
+SET(CMAKE_SYSTEM_NAME Generic)
+SET(CMAKE_C_COMPILER avr-gcc)
+SET(CMAKE_CXX_COMPILER avr-g++)
+SET(CMAKE_CAT cat)
+
+SET(CSTANDARD "-std=gnu11")
+SET(CPPSTANDARD "-std=gnu++11")
+
+SET(WARN "-Wall -Wextra")
```

https://cmake.org/cmake/help/latest/command/target_compile_options.html

---

It's practical to turn this into an `option`
https://cmake.org/cmake/help/v3.13/command/option.html

```suggestion
option(ENABLE_WARNINGS "compile with warnings enabled" ON)
if(ENABLE_WARNINGS)
    ...
else(ENABLE_WARNINGS)
    target_compile_options(... ... -w)
endif(ENABLE_WARNINGS)
```